### PR TITLE
[E2E] Tests for version: verifying semver

### PR DESCRIPTION
**Pull request by jchavezlavalle**
**Description of the Pull Request (PR):**

E2E Tests for version: verification of semantic version.

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: e2e/version/version.go [hunk 1]
```diff
@@ -57,13 +60,17 @@ func testEqualVersion(t *testing.T) {
 			if err != nil {
 				t.Fatalf("Failed to obtain version: %+v", err)
```
Comment (WestleyK) [inline e2e/version/version.go:61]: Should you print the output here:

```go
t.Log(string(out))
```

File: e2e/version/version.go [hunk 2]
```diff
@@ -38,8 +39,10 @@ func testSemanticVersion(t *testing.T) {
...
 			}
-
-			if semanticVersion, err := semver.Make(string(out)); err != nil {
+			outputVersion := strings.Replace(string(out), "singularity version ", "", -1)
```
Comment (ikaneshiro) [inline e2e/version/version.go]: I think `strings.TrimPrefix()` is a better util function to use here

File: e2e/version/version.go [hunk 3]
```diff
@@ -38,8 +39,10 @@ func testSemanticVersion(t *testing.T) {
...
+			outputIndex := strings.Index(string(out), "-")
```
Comment (ikaneshiro) [inline e2e/version/version.go]: Why are you splitting the version here? It seems like `semver` will properly handle the build metadata?

**Review by ikaneshiro — CHANGES_REQUESTED** [review]

File: e2e/version/version.go (hunk 3 above)
Comment (jchavezlavalle) [reply e2e/version/version.go]: No it doesn't, that's why I'm just considering the version with this library until the patch to verify the version. `singularity --version` outputs : `singularity version 3.2.0-541.g4e601d8d9` so from the `-` to the end, this library does not detect the tag (it just handles major version, minor version and patch) Also, consider that `singularity version` (command) gives as output: `3.2.0-541.g4e601d8d9` so if I need to consider if both versions are the same, I need to split the `singularity version` string  if present, which is the output from `singularity --version`

**Review by ikaneshiro — APPROVED** [review]
LGTM

**Pull request merged**